src/engines/kotor/gui/ingame/hud.cpp

```diff
@@ -81,49 +209,60 @@ HUD::HUD(Module &module, Engines::Console *console)
 		if (it->width == wWidth && it->height == wHeight)
 			foundRes = &*it;
 
+	bool scale = false;
 	if (!foundRes) {
-		Common::UString resString;
-
 		for (std::set<Resolution>::const_iterator it = availableRes.begin(); it != availableRes.end(); ++it) {
-			if (!resString.empty())
-				resString += ", ";
-
-			resString += Common::UString::format("%dx%d", it->width, it->height);
+			// Search for an existing interface with 800x600 resolution with preference to the pc interface.
+			if (Common::UString(it->gui) == "mipc28x6" || (Common::UString(it->gui) == "mi8x6" && !foundRes)) {
```

Comment:
Why are you comparing the name instead width and height directly?

You can also just break when one is found, instead of checking that foundRes is not assigned yet.

I.e. I think that
```
if ((it->width == 800) && (it->height == 600)) {
	foundRes = &*it;
	break;
}
```

would be way clearer.

Reply:
(In C++11, I'd even say that's a std::find() with a lambda checking width and height, but that's future talk :P)

Reply:
Since the pc version of kotor also contains the gui file for the xbox ui, i wanted to prefer the pc ui and the only way of doing this, is by comparing their names. Maybe later you can add a command line switch for forcing the xbox ui. Also i wanted to make sure, that in the case of both being found and supported, the pc version is used since that would be what a normal pc user would expect first and, as far as i know, the pc ui is not contained in the xbox game but the xbox ui is contained in the pc game.

Reply:
That's why availableRes is a std::set, operator<() only checks the dimensions, and the PC version is checked first. availableRes doesn't contain the xbox GUI file in PC version, even though the file is theoretically available.

Reply:
(Yes, that's a bit of a hack in either case, I know)

Reply:
you're right, i will correct this